archivy/models.py

```diff
@@ -1,23 +1,57 @@
-import datetime
+from datetime import datetime
+from typing import List, Optional
 from urllib.parse import urljoin
 
-import validators
-import requests
-import html2text
 import frontmatter
-from flask import flash
+import html2text
+import requests
+import validators
+from attr import attrs, attrib
+from attr.validators import instance_of, optional
 from bs4 import BeautifulSoup
 from flask import current_app
+from flask import flash
 
 from archivy import extensions
-from archivy.search import add_to_index
 from archivy.data import create
+from archivy.search import add_to_index
 
 
+# TODO: use this as 'type' field
+# class DataobjType(Enum):
+#     BOOKMARK = 'bookmark'
+#     POCKET_BOOKMARK = 'bookmark imported from pocket'
+#     NOTE = 'note'
+#     PROCESSED_DATAOBJ = 'bookmark that has been processed'
+
+@attrs(kw_only=True)
 class DataObj:
     __searchable__ = ["title", "content", "desc", "tags"]
 
+    id: Optional[int] = attrib(validator=optional(instance_of(int)),
```

Comment:
I like this clear declaration of the attrs as it provides a clearer structure.

Reply:
Does this totally eliminate the need for a `init` function and instead simply maps the keyword arguments if they exist?

Reply:
Yes, attts creates the constructor. With `kw_only=True`, all parameters of the constructor are keyword arguments.